lib/logstash/inputs/cloudwatch_logs.rb

```diff
@@ -110,16 +110,21 @@ def check_start_position_validity
   public
   def run(queue)
     @queue = queue
+    @priority = []
     _sincedb_open
     determine_start_position(find_log_groups)
 
     while !stop?
-      groups = find_log_groups
-
-      groups.each do |group|
-        @logger.debug("calling process_group on #{group}")
-        process_group(group)
-      end # groups.each
+      begin
+        groups = find_log_groups
```

Comment:
I'm wondering if I should move this line out of your begin/rescue here and add back in the handling for find_log_groups that was part of 12565715f7752de08a99e7eabc184ab87a5f62d3. 

I think that if we got throttled fetching log groups here - we might get into a situation where we're always being throttled, and just stopping.

It's probably a very rare corner case... but might be better to handle it differently than how we handle an interruption while processing a group.